### PR TITLE
KAFKA-9546 Allow custom tasks through configuration

### PR DESCRIPTION
**Pull request by gcsaba2**
Currently FileStreamSourceConnector can only return a task of type FileStreamSourceTask. With this change the users can override it and provide a custom task class via configuration.

Testing was done via unit tests. There's one positive case (custom Task class provided through config) and one negative (invalid class java.io.File was provided). The already existing unit tests are testing the default behavior, when FileStreamSourceTask is used.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by rhauch** [comment]
As noted on https://issues.apache.org/jira/browse/KAFKA-9546, I'm not convinced this is useful change, since it would complicate these example connectors that ship with AK and would make customizing them very impractical from a build, release, and installation perspective.

**Comment by rhauch** [comment]
Thanks for the PR, @gcsaba2. However, I'm going to close this per my comments on https://issues.apache.org/jira/browse/KAFKA-9546. 

TLDR; the `FileStreamSourceConnector` class is intentionally a simple example of a connector, and extending this as suggested makes that it harder to use and more error prone. Instead, just create your own connector with the desired functionality.
